### DISCUSSION TITLE
Permanent Labels on Map?

### DISCUSSION THREAD

**Discussion by hm535** [foxglove/community#979] [Q&A]
Hi everyone,
I am trying to display some polygons on the Map Panel, with some text labels (the text contains changing values based on sensor readings).  So far, I have been able to add the polygons to the map using foxglove GeoJSON messages, and through "properties.name" I can add a pop-up text label.  However the label only popups when the cursor is hovering over a specific polygon; and since I need to display multiple polygons at the same time, it is not possible for me to hover over all the polygons simultaneously.  Is there anyway to make the text labels show up on the map permanently, ideally through the GeoJSON messages?   Since the text values are changing based on sensor readings, it would be best if I could send something like this: 

```
 sample_polygon = {
         'type': "FeatureCollection",
         'features': [{
             'type': "Feature",
             'properties': {
                 'name': "Named Polygon",
                 **'labelPermanent': True,**
                 'style': {
                 'color': "red"
                 },
             },
             'geometry': {
                 'type': 'Polygon',
                 "coordinates": [
                     [
                       ## bunch of coordinates
                     ]
                 ]
             }
         }]
     }
```

**Comment by defunctzombie** [comment]
Is there any aspect of GeoJSON that provides for text display? I am not aware of such a thing so we'd need to see what custom properties we could introduce for such a label. Our current GeoJSON support comes from what leaflet supports (https://leafletjs.com/examples/geojson/)

**Reply by hm535** [reply]
Ahh, I don't think GeoJSON has an aspect for text display.  But leaflet provides for a "permanent" bind tool tip option which does what I am looking for: [https://leafletjs.com/reference.html#tooltip](https://leafletjs.com/reference.html#tooltip.)

From what I can tell, Foxglove does use the tooltip feature to display "properties.name", so it might just be a matter of toggling the "permanent" option between True/ False.

Would there be a way to inherit from the Map panel, when building a custom extension?  I'm not experienced in typescript/ javascript enough to build an entire Map extension from scratch; but if I can inherit from the existing Map panel, I should be able to over-write this specific property for the text display.

**Reply by defunctzombie** [reply]
Unfortunately there is no way to inherit from existing panels.

From what I understand - your ask is you introduce a new property within the feature that could specify if that particular feature's label should always be displayed? Or are you wanting a panel setting?

Something I want to note as a potential gotcha with such a setting (in the messages or in the panel) is that when loading a local file we show all of the available points so maybe this should be something that triggers only for the current point? Or would you expect to see your label for all historic points as well?
